sdk/resourcemanager/Azure.ResourceManager.Core/src/Operation/ArmOperation.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// Abstract class for long-running or synchronous applications.
+    /// </summary>
+    public abstract class ArmOperation : Operation
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ArmOperation"/> class for mocking.
+        /// </summary>
+        protected ArmOperation()
+        {
+        }
+
+        /// <summary>
+        /// Waits for the completion of the long running operations.
+        /// </summary>
+        /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="CancellationToken.None" />. </param>
+        /// <returns> A response with the <see cref="ArmOperation{TOperations}"/> operation for this resource. </returns>
+        /// <remarks>
+        /// <see href="https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-longrunning">Details on long running operation object.</see>
+        /// </remarks>
+        public virtual Response WaitForCompletion(CancellationToken cancellationToken = default)
+        {
+            return WaitForCompletion(OperationInternals<Response>.DefaultPollingInterval.Seconds, cancellationToken);
+        }
+
+        /// <summary>
+        /// Waits for the completion of the long running operations.
+        /// </summary>
+        /// <param name="pollingInterval"> The polling interval in seconds to check for status. </param>
```

Comment:
Polling interval is in milliseconds, see: https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.delay?view=net-5.0#System_Threading_Tasks_Task_Delay_System_Int32_System_Threading_CancellationToken_

Reply:
changed this to TimeSpan to be consistent with WaitForCompletionAsync